### PR TITLE
Upgrade to node v16/npm v8

### PR DESCRIPTION
**Pull request by mofojed**
- Update to node v16 (lts/gallium)
- Removed all package-lock.jsons, then ran `npm install` and `lerna exec npm install -- --package-lock-only`
- Pinned some eslint dependencies
- Updated some exports that were breaking tests importing from `/dist/` folders directly
- Update engine restrictions
- Update GitHub actions to use node v16

### CODE REVIEW

File: packages/icons/package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
   },
   "peerDependencies": {
     "@fortawesome/fontawesome-svg-core": "^1.2.34",
-    "@fortawesome/react-fontawesome": "^0.1.14"
+    "@fortawesome/react-fontawesome": "^0.1.12"
```
Comment (vbabich) [inline packages/icons/package.json:46]: Why downgrade?

File: packages/dashboard-core-plugins/src/panels/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ export { default as CommandHistoryPanel } from './CommandHistoryPanel';
 export { default as ConsolePanel } from './ConsolePanel';
 export { default as DropdownFilterPanel } from './DropdownFilterPanel';
 export { default as FileExplorerPanel } from './FileExplorerPanel';
+export { FilterSet, default as FilterSetManager } from './FilterSetManager';
```
Comment (vbabich) [inline packages/dashboard-core-plugins/src/panels/index.js:6]: Getting compile error here
```
Attempted import error: 'FilterSet' is not exported from './FilterSetManager.js'.
```

File: packages/embed-grid/package.json [hunk 3]
```diff
@@ -26,12 +26,12 @@
     "@deephaven/react-hooks": "^0.11.0",
     "@deephaven/utils": "^0.11.0",
     "@fortawesome/fontawesome-svg-core": "1.2.36",
-    "@fortawesome/react-fontawesome": "^0.1.15",
+    "@fortawesome/react-fontawesome": "^0.1.12",
```
Comment (mattrunyon) [inline packages/embed-grid/package.json:29]: Downgrade? I think this is the same thing Vlad commented on, just in a different package.json

**Review by mattrunyon — CHANGES_REQUESTED** [review]
Main readme needs an update (references node 14)
Should add a `.nvmrc` so we can all set up auto switching if we wanted to [link](https://github.com/nvm-sh/nvm#deeper-shell-integration)

File: packages/icons/package.json (hunk 1 above)
Comment (mofojed) [reply packages/icons/package.json:46]: Accident

File: packages/embed-grid/package.json (hunk 3 above)
Comment (mofojed) [reply packages/embed-grid/package.json:29]: Accident

**Review by mattrunyon — APPROVED** [review]

**Review by vbabich — DISMISSED** [review]

**Review by mattrunyon — DISMISSED** [review]

**Review by vbabich — COMMENTED** [review]
I'm getting a white screen and this error in the console when trying to start the app:
```
Uncaught Error: Cannot find module 'react-refresh/runtime'
    at webpackMissingModule (ReactRefreshEntry.js:6:1)
    at Object.../../node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js (ReactRefreshEntry.js:6:1)
    at __webpack_require__ (bootstrap:851:1)
    at fn (bootstrap:150:1)
    at Object.1 (main.chunk.js:22:1)
    at __webpack_require__ (bootstrap:851:1)
    at checkDeferredModules (bootstrap:45:1)
    at Array.webpackJsonpCallback [as push] (bootstrap:32:1)
    at main.chunk.js:1:97
```

**Comment by mofojed** [comment]
Looking into it

**Review by vbabich — APPROVED** [review]

**Pull request merged**